### DISCUSSION TITLE
What meta info should we show on a track page

### DISCUSSION THREAD

**Discussion by iHiD** [exercism/v3#2276] [General]
This issue is about the new version of the unjoined track page (e.g. https://exercism.io/tracks/ruby in incognito).

We are adding various information about the language features. We also want to add information about **how the language applies to the real world**. This discussion is about the later, not the former.

I think there are three lose groups that are relevant:
- Learning the language
- The use of the language
- The "value" of the language to a person.

For example, for the "learning the language" bit we might show something like:
![94564375-d2664b00-025f-11eb-8a91-10330bc89fc2](https://user-images.githubusercontent.com/286476/94583584-8920f600-0275-11eb-9f5f-f78cd1c0c8e5.png)

What information would you actually find useful when deciding what track to start next on Exercism. Examples might include:
Using this as inspiration, what kind of things are important in deciding what language to learn?

- Average salary
- Trending (usage)
- Developer supply/demand
- Where the language is used (e.g. mobile, web, integrated systems)
- % of people on Exercism that are trying it
- Any sort of GitHub stats.

What would **you** personally find helpful?

**Comment by bobahop** [comment]
Even before I retired I would dabble with a language if I thought it might yield an approach not evident in my current language/s. For example, I learned about regular expressions from dabbling in Perl and tcl. Only then did I look for a way to use regexp in classic VB, which I found by referencing a library in IE. Dabbling in F# and Scala led me to to use functional programing more in C#. So a description of what a language uniquely has to offer is what I would personally find useful.

**Reply by ynfle** [reply]
To add to @bobahop, I think the value could be related to someone not being exposed to a language that has similar concepts or even that the candidate language brings them out further. For example OOP in Python versus Ruby, where both have it but Ruby is much stronger and brings out the concept much more

**Reply by iHiD** [reply]
Thanks. These fit into the "information about the language features" bit, which is an extra thing. For the context of this discussion I'm specifically interested in the things away from language features - specifically how it applies to the real world.

**Reply by iHiD** [reply]
That said, a good example of how this might work in the context of "real world" is something along the lines of "widen your skill base by learning this language, which nicely compliments what you know" (which is maybe what you're both getting at?)

**Reply by bobahop** [reply]
Yes, learning programming techniques is like investing: it's good to diversify. As for "real word" relevance, I notice TIOBE, StackOverflow ratings of popular and/or "loved" languages. I use google trends to compare languages. I use Dice to see what jobs are being offered for what languages. Perhaps Exercism can be a portal for that kind of info. Not only might it aggregate the info, but it could also point to the sources as well.

**Reply by iHiD** [reply]
Yeah - I really like the idea of having a little microsite for this as a standalone feature: "Which language should I try next?"

**Comment by SaschaMann** [comment]
> What information would you actually find useful when deciding what track to start next on Exercism

Not sure if this is any helpful, but: none. I don't choose what language to learn on Exercism, I already have a language in mind and then check if there's a track on Exercism.

**Reply by iHiD** [reply]
Yep. I think that's true for a lot of more experienced developers who understand the wider ecosystem. For more junior ones I think it's hard to know what to learn next and I think we can add value here, as there's not necessarily a good place to go to find that out (like, we could spin off a mini-site specifically for "Choose your next language" on the back of this content.

**Reply by SaschaMann** [reply]
If I look at the websites of programming languages, which _are_ trying to convince you to use them, they use none of the metrics you gave as an example, but are closer to what we currently have: a list of features and common use-cases.

E.g. https://nim-lang.org/, https://www.rust-lang.org/, https://julialang.org/, https://golang.org/ (ok, this has nothing, but I guess "Google made it" is sufficient), https://www.python.org/, https://clang.llvm.org/.

**Comment by ErikSchierboom** [comment]
Like Sascha, I personally don't care about a language being trending, the supply/demand or the average salary. I mainly choose languages based on what I can learn from them, although their primary use (web-based, cross-platform, etc.) is also something I consider. That said, I can imagine this being relevant to people looking for new jobs or just starting out with programming (although that is not our target audience at this time).

My main "issue" with some of the suggested metrics is that they can be really hard to get right. Suppose one wants to display how many jobs there are available for a language. The first thing to note is that this is of course very dependent on where the student lives. Some countries use certain languages far more than other, so we'd have to take this into account. Secondly, trending is very hard to quantify for languages. Various metrics have been attempted (like # of SO posts or Google search results), but I feel like none are a really good indicator. Does a language have more SO posts because it is popular, or because it is a confusing language? The "% of people on Exercism that are trying it" at least has the benefit that we can actually measure this precisely, while there is the minor "risk" of funneling people into the popular languages.

**Reply by SaschaMann** [reply]
> My main "issue" with some of the suggested metrics is that they can be really hard to get right.

Do you know of any single source that "gets it right"? Most of the popular rankings, surveys etc. all have wildly varying and often oversimplified results.

**Reply by ErikSchierboom** [reply]
I don't think any of the popular rankings or surveys actually get it right. There is always a lot of bias and usually a relatively small sample size. And then you get posts like this: https://stackoverflow.blog/2017/06/15/developers-use-spaces-make-money-use-tabs/ 🤦

**Reply by bobahop** [reply]
Whether they get it right nor not, the indexes and surveys keep getting media coverage because many people either like the validation of having their favorite language  high in the rankings or they like to criticize the methodology when their favorite language isn't as high in the rankings as they think it should be. Accurate or not, a mention in an index or survey may be the first place someone may hear about a language. It's like Oscar Wilde said "There is only one thing in life worse than being talked about, and that is not being talked about."

I want to say that I first heard about Rust via the StackOverflow "most-loved" survey, but I may actually have first heard of it on Quora when reading about C++.

**Reply by bobahop** [reply]
If we used the "% of people on Exercism that are trying it", how would that be calculated? How would inactivity be factored in? I wonder at the accuracy of current Exercism metrics, when Rust is supposed to have 45 mentors. Unfortunately, I see no evidence of that in the weekly stats. :(

**Comment by kotp** [comment]
On 20/09/30 08:35AM, Bob Hoeppner wrote:
> If we used the "% of people on Exercism that are trying it", how would

We can surely avoid the harder to quantify "are trying it" and instead
report on "have tried it" by counting those that have at least made a
submission.

**Reply by bobahop** [reply]
That sounds better. And what would constitute having tried it? How many completed exercises on the track count as having tried it would be the criteria to determine.

**Reply by bobahop** [reply]
I mean, I'm not sure that just submitting "Hello World" should count.

**Reply by ynfle** [reply]
We could have a separate metric which is "showed interest in". One can be for completed `hello-world` and attempted another exercise and one for completed more than 2 after `hello-world`
